lib/travis/api/v3/models/repository.rb

```diff
@@ -40,9 +40,14 @@ def migrated_at
     end
 
     def slug
-      @slug ||= "#{owner_name}/#{name}"
+      @slug ||= "#{owner.login}/#{name_from_url}"
     end
 
+    def name_from_url
+      return url.split('/')[1].gsub('.git', '') if vcs_type == 'BitbucketRepository' && url && url.split('/')[1]
```

Comment:
we shouldn't compare vcs type in api - such things should be solved by vcs service.I have 2 propositions:
1) Call to vcs service and cache 
2) Add additional field to repository